packages/sdk/src/streamUtils.ts

```diff
@@ -6,7 +6,53 @@ import {
 } from '@river-build/proto'
 import { ParsedEvent, ParsedMiniblock } from './types'
 import { bin_toHexString } from '@river-build/dlog'
-import { isDefined } from './check'
+import { isDefined, logNever } from './check'
+
+export function isPersistedEvent(event: ParsedEvent, direction: 'forward' | 'backward'): boolean {
+    if (!event.event) {
+        return false
+    }
+
+    switch (event.event.payload.case) {
+        case 'channelPayload':
+            return true
+        case 'dmChannelPayload':
+            return true
+        case 'gdmChannelPayload':
+            return true
+        case 'mediaPayload':
+            return true
+        case 'userPayload':
+            return direction === 'forward' ? true : false
+        case 'userSettingsPayload':
+            return direction === 'forward' ? true : false
+        case 'miniblockHeader':
+            return true
+        case 'userMetadataPayload':
+            return direction === 'forward' ? true : false
+        case 'memberPayload': {
+            switch (event.event.payload.value.content.case) {
+                case 'keySolicitation':
+                    return direction === 'forward' ? true : false
+                case 'keyFulfillment':
+                    return direction === 'forward' ? true : false
+                case undefined:
+                    return false
+                default:
+                    return direction === 'forward' ? true : false
+            }
+        }
+        case 'spacePayload':
+            return direction === 'forward' ? true : false
+        case 'userInboxPayload':
+            return direction === 'forward' ? true : false
```

Comment:
we do - the only reason we save events on moving forward is so that when we load up a block from persistence "from the last snapshot" it properly rolls up all the events since the last snapshot to re-create the proper state. I am going to work on something to fix this but it will take more time.